basic_pitch/note_creation.py

```diff
@@ -240,16 +249,31 @@ def note_events_to_midi(
         pitch_bend_midi_ticks[pitch_bend_midi_ticks > N_PITCH_BEND_TICKS - 1] = N_PITCH_BEND_TICKS - 1
         pitch_bend_midi_ticks[pitch_bend_midi_ticks < -N_PITCH_BEND_TICKS] = -N_PITCH_BEND_TICKS
         for pb_time, pb_midi in zip(pitch_bend_times, pitch_bend_midi_ticks):
-            instrument_per_piano_key[note_number].pitch_bends.append(pretty_midi.PitchBend(pb_midi, pb_time))
-
-    for inst in instrument_per_piano_key.values():
-        # only add instruments with notes
-        if len(inst.notes) > 0:
-            mid.instruments.append(inst)
+            instrument.pitch_bends.append(pretty_midi.PitchBend(pb_midi, pb_time))
+    mid.instruments.extend(instruments.values())
 
     return mid
 
 
+def drop_overlapping_pitch_bends(
+    note_events_with_pitch_bends: List[Tuple[float, float, int, float, Optional[List[int]]]]
+) -> List[Tuple[float, float, int, float, Optional[List[int]]]]:
+    """Drop pitch bends from any notes that overlap in time with another note
+
+    TODO: naive N^2 implementation!
+    """
+    note_events = list(map(list, note_events_with_pitch_bends))  # tuple to (mutable) list
+    for i, note in enumerate(note_events):
+        start_time, end_time = note[:2]
+        for other_note in note_events[:i]:
+            other_start_time, other_end_time = other_note[:2]
+            if start_time < other_end_time and end_time > other_start_time:
+                note[-1] = None  # last field is pitch_bend
+                other_note[-1] = None
```

Comment:
```suggestion
    note_events = sorted(note_events)
    for i, note in enumerate(note_events)[:-1]:
        start_time, end_time = note[:2]
        for other_note in note_events[i+1:]:
            other_start_time, other_end_time = other_note[:2]
            if other_start_time > end_time:
                break
            if start_time < other_end_time and end_time > other_start_time:
                note[-1] = None  # last field is pitch_bend
                other_note[-1] = None
```

Reply:
Given the complexity of this logic, could you add a test?